src/ng-select/items-list.ts

```diff
@@ -179,12 +182,29 @@ export class ItemsList {
 
     private _getDefaultFilterFunc(term: string) {
         return (option: NgOption) => {
-            return searchHelper.stripSpecialChars(option.label || '')
+            let found: number;
+            found = searchHelper.stripSpecialChars(option.label || '')
                 .toUpperCase()
-                .indexOf(searchHelper.stripSpecialChars(term).toUpperCase()) > -1;
+                .indexOf(searchHelper.stripSpecialChars(term).toUpperCase());
+            if (found > -1) {
+                this._highlightLabelWithSearchTerm(option, term, found);
+            }
+            return found > -1;
         };
     }
 
+    private _highlightLabelWithSearchTerm(option: NgOption, term: string, indexOfTerm: number) {
+        if (indexOfTerm > -1 && !option.selected && !option.disabled) {
```

Comment:
It seems to add unneeded characters in some cases.
![image](https://user-images.githubusercontent.com/3028012/35777365-60fb9614-09b5-11e8-8024-a28895945922.png)

Reply:
fixed the end index was wrong on the third substring